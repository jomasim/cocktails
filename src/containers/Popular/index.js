import CockTail from '../../components/CockTail'
import './popular.css'
const Popular = ({ data, showCockTail }) => {
  return (
    <div style={{ marginTop: '50px' }}>
      <div style={{ textAlign: 'center' }}>
        <p className='popular-tit'>Popular</p>
      </div>
      <div
        className='co-co'
        style={{
          display: 'flex',
          marginTop: '60px',
          flexWrap: 'wrap',
          justifyContent: 'center'
        }}
      >
        {data.map(item => (
          <CockTail
            key={item.idDrink}
            data={item}
            showCockTail={showCockTail}
          />
        ))}
      </div>
    </div>
  )
}

export default Popular
