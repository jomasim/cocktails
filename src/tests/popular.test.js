import React from 'react'
import Popular from '../containers/Popular'
import { shallow } from 'enzyme'

const data = [
  {
    idDrink: '178326',
    strDrink: 'White Wine Sangria',
    strCategory: 'Punch / Party Drink',
    strGlass: 'Pitcher'
  },
  {
    idDrink: '178326',
    strDrink: 'White Wine Sangria',
    strCategory: 'Punch / Party Drink',
    strGlass: 'Pitcher'
  }
]
const showCockTail = jest.fn()
it('should render  popular container ', () => {
  const wrapper = shallow(<Popular data={data} showCockTail={showCockTail} />)
  expect(wrapper.length).toBe(1)
})
it('matches snapshot', () => {
  const wrapper = shallow(<Popular data={data} showCockTail={showCockTail} />)
  expect(wrapper).toMatchSnapshot()
})
