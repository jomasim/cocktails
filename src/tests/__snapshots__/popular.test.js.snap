// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "marginTop": "50px",
    }
  }
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <p
      className="popular-tit"
    >
      Popular
    </p>
  </div>
  <div
    className="co-co"
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginTop": "60px",
      }
    }
  >
    <CockTail
      data={
        Object {
          "idDrink": "178326",
          "strCategory": "Punch / Party Drink",
          "strDrink": "White Wine Sangria",
          "strGlass": "Pitcher",
        }
      }
      key="178326"
      showCockTail={[MockFunction]}
    />
    <CockTail
      data={
        Object {
          "idDrink": "178326",
          "strCategory": "Punch / Party Drink",
          "strDrink": "White Wine Sangria",
          "strGlass": "Pitcher",
        }
      }
      key="178326"
      showCockTail={[MockFunction]}
    />
  </div>
</div>
`;
